backend/instance.ts

```diff
@@ -86,12 +87,12 @@ export class Instances {
     });
   }
 
-  add(): Instance {
-    this.currentPort++;
-    const port = this.currentPort;
+  async add(): Promise<Instance> {
+    const port = await detectPort(this.currentPort + 1);
     if (this.instances.has(port)) {
       throw new Error(`Already have Webxdc instance at port: ${port}`);
     }
```

Comment:
I think this if-statement is redundant now since `detectPort()` will return a valid port that's not taken. But I'm fine with leaving it here for good measure.